utils/coreutils/techutils.go

```diff
@@ -87,27 +94,34 @@ func DetectTechnologies(path string, isCiSetup, recursive bool) (map[Technology]
 	if err != nil {
 		return nil, err
 	}
-	detectedTechnologies := make(map[Technology]bool)
-	for _, file := range filesList {
-		techNames := detectTechnologiesByFile(strings.ToLower(file), isCiSetup)
-		for _, techName := range techNames {
-			detectedTechnologies[techName] = true
-		}
-	}
+	detectedTechnologies := detectTechnologiesByFilePaths(filesList, isCiSetup)
 	return detectedTechnologies, nil
 }
 
-func detectTechnologiesByFile(file string, isCiSetup bool) (detected []Technology) {
-	detected = []Technology{}
-	for techName, techData := range technologiesData {
-		if !isCiSetup || (isCiSetup && techData.ciSetupSupport) {
-			for _, indicator := range techData.indicators {
-				if strings.Contains(file, indicator) {
-					detected = append(detected, techName)
+func detectTechnologiesByFilePaths(paths []string, isCiSetup bool) (detected map[Technology]bool) {
+	detected = make(map[Technology]bool)
+	exclude := make(map[Technology]bool)
+	for _, path := range paths {
+		for techName, techData := range technologiesData {
+			if !isCiSetup || (isCiSetup && techData.ciSetupSupport) {
+				for _, excludeFile := range techData.exclude {
+					if strings.Contains(path, excludeFile) {
+						exclude[techName] = true
+					}
+				}
+				if _, exist := exclude[techName]; !exist {
+					for _, indicator := range techData.indicators {
+						if strings.Contains(path, indicator) {
```

Comment:
Wouldn't it be better and more accurate to use `HasSuffix` instead of `Contains` here?